### PR TITLE
Fix a few warnings

### LINKED ISSUE #13: Compiler Warning: Unused Result
Compiling BuDDy on Ubuntu 22.04 prints the following warning.
```
src/bddio.c:499:7: warning: ignoring return value of ‘fscanf’
                   declared with attribute ‘warn_unused_result’
                   [-Wunused-result]
  499 |       fscanf(ifile, "%d", root);
```

### PR DESCRIPTION
**Pull request by nhusung**
A few warnings were related to `lexer.cxx` and `parser.cxx` (in `bddcalc`) using the `register` keyword. These files were generated by old versions of Flex and Bison. I thought it would be better to remove these files and generate them again.

Also closes #13

### CODE REVIEW

File: examples/bddcalc/runtest.in [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @CMAKE_CURRENT_BINARY_DIR@/bddcalc $f \
+        | grep -E -v '^Garbage collection' >> @CMAKE_CURRENT_BINARY_DIR@/result
+done
+
+diff expected @CMAKE_CURRENT_BINARY_DIR@/result
+if [ $? == 0 ]; then
+    echo "Success."
+    exit 0
+else
+    echo "Fail. Differences as show above (left: expected, right: actual)"
```
Comment (ssoelvsten) [inline examples/bddcalc/runtest.in]: "as show**n** above" (with an *n*)

**Review by ssoelvsten — COMMENTED** [review]
Thanks! Except for a typo on the last commit, I don't notice anything - I also expect you to already have tested it works.

File: examples/bddcalc/runtest.in (hunk 1 above)
Comment (nhusung) [reply examples/bddcalc/runtest.in]: Thanks, good catch. Seems like I was too tired on train yesterday 😴

**Comment by nhusung** [comment]
Regarding testing: I successfully ran the `runtest` script (`$builddir/examples/bddcalc/runtest`) and a few BDD Benchmark items. I did not test any reordering features yet, but to me, it seems that 052eb97a23b4fa80e38689fa347dcf8ed70702e9 fixes an actual bug. A function pointer is never null, so there never was an early return in case BuDDy was not ready for reordering.

**Pull request merged**